flint/flint-spark-integration/src/main/scala/org/opensearch/flint/spark/FlintSpark.scala

```diff
@@ -227,18 +230,30 @@ object FlintSpark {
      * @return
      *   index builder
      */
-    def addPartitionIndex(colNames: String*): IndexBuilder = {
+    def addPartitions(colNames: String*): IndexBuilder = {
       colNames
-        .map(colName =>
-          allColumns.getOrElse(
-            colName,
-            throw new IllegalArgumentException(s"Column $colName does not exist")))
-        .map(col =>
-          new PartitionSkippingStrategy(columnName = col.name, columnType = col.dataType))
+        .map(findColumn)
+        .map(col => PartitionSkippingStrategy(columnName = col.name, columnType = col.dataType))
         .foreach(indexedCol => indexedColumns = indexedColumns :+ indexedCol)
       this
     }
 
+    /**
+     * Add value set skipping indexed column.
+     *
+     * @param colName
+     *   indexed column name
+     * @return
+     *   index builder
+     */
+    def addValueSet(colName: String): IndexBuilder = {
```

Comment:
what if user specify the col as partitionIndex and valueSet index? the later one will override the former one?

Reply:
yes, the validation is missing and there will be 2 columns of same name in the index building DF (if Spark allows this).
Will add the check to prevent this and we may support this later if there is requirement. Thanks!

Reply:
Fixed dup column check with IT in https://github.com/opensearch-project/sql/pull/1708/commits/4992ad97da114b2fa00274412b98e474acfc9d61